railties/test/application/rake_test.rb

```diff
@@ -126,6 +126,47 @@ def test_rake_routes_with_controller_environment
       assert_equal "Prefix Verb URI Pattern     Controller#Action\n  cart GET  /cart(.:format) cart#show\n", output
     end
 
+    def test_rake_routes_with_controller_environment_is_deprecated
```

Comment:
Why? We still have this functional, and I think that we need remove this test after remove `CONTROLLER` env variable.